api/provider/aws/task_list.go

```diff
@@ -0,0 +1,94 @@
+package aws
+
+import (
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+func (t *TaskProvider) List() ([]models.TaskSummary, error) {
+	clusterNames, err := listClusterNames(t.AWS.ECS, t.Config.Instance())
+	if err != nil {
+		return nil, err
+	}
+
+	summaries := []models.TaskSummary{}
+	for _, clusterName := range clusterNames {
+		runningTaskARNs, err := t.listTaskIDs(clusterName, ecs.DesiredStatusRunning)
+		if err != nil {
+			return nil, err
+		}
+
+		stoppedTaskARNs, err := t.listTaskIDs(clusterName, ecs.DesiredStatusStopped)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, taskID := range append(runningTaskARNs, stoppedTaskARNs...) {
+			summary := models.TaskSummary{
+				TaskID: taskID,
+			}
+			summaries = append(summaries, summary)
+		}
+
+	}
+
+	if err := t.populateTaskSummaries(summaries); err != nil {
+		return nil, err
+	}
+
+	return summaries, nil
+}
+
+func (t *TaskProvider) listTaskIDs(clusterName, status string) ([]string, error) {
+	taskIDs := []string{}
+	fn := func(output *ecs.ListTasksOutput, lastPage bool) bool {
+		for _, arn := range output.TaskArns {
+			// task arn format: arn:aws:ecs:region:012345678910:task-definition/taskID:revision
```

Comment:
This is an arn for a `task definition`, not a `task`. 

The  [AWS API Docs](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_DescribeTasks.html) show task arns to be in the format:  `"arn:aws:ecs:us-east-1:012345678910:task/1dc5c17a-422b-4dc4-b493-371970c6c4d6"`

We should be able to get away with just: 
```
taskID := strings.Split(aws.StringValue(arn), "/")[1]
```